libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -250,11 +251,45 @@ const luaL_Reg AP_Logger_functions[] = {
     {NULL, NULL}
 };
 
+static int AC_Fence_Load(lua_State *L) {
+    AC_Fence * fence = AC_Fence::get_singleton();
+    if (fence == nullptr) {
+        return luaL_argerror(L, 1, "fence not supported on this firmware");
+    }
+
+    const uint8_t args = lua_gettop(L) - 1;
+    if (args < 3) {
+        return luaL_argerror(L, args, "requires at least 3 arguments");
+    }
+
+    class AC_PolyFenceItem new_items[args];
+
+    for (uint8_t i=0; i<args; i++) {
+        AC_PolyFenceItem *item = check_AC_PolyFenceItem(L,i-args);
```

Comment:
I have no idea why we have to use negative stack index here where as in the logger binding there positive. But this does seem to work as expected. We start at larger negative number and work up to -1 in order to get the fence to come out in the expected order.